Added support for cmake / conan

massenz opened the pull request:
Hi,

thanks for putting this together - much appreciated!

I thought I'd contribute back by adding support for CMake & adding `conan` (see http://conan.io ) so that you could remove the OpenSSL git submodule.

BTW - the `example` runs into a weird `parse error` when used against google.com, but it works just fine when used against my [own server](https://github.com/massenz/distlib) and returns the correct JSON content - not sure why.

Commit:
Added support for cmake / conan.

Commit:
Updated README

ivere27 commented:
Hi,

thanks for the contributes and using the library.
it's better to leave the openssl in git submodule. for someone who doesn't use the conan.

I'll take a look at the bug(parsing error). is it 'http://google.com' or  ssl/tls 'https://google.com'?

ivere27 commented on CMakeLists.txt:
```diff
@@ -0,0 +1,49 @@
+# Version number
+set(RELEASE_MAJOR 0)
+set(RELEASE_MINOR 2)
+set(RELEASE_PATCH 0)
```
let's make same as

#define SIMPLE_HTTP_REQUEST_VERSION_MAJOR 0
#define SIMPLE_HTTP_REQUEST_VERSION_MINOR 1
#define SIMPLE_HTTP_REQUEST_VERSION_PATCH 0

massenz replied:
Ok - updated that, but the reason was that I saw you'd already released `0.1.0`, so my assumption was that any subsequent build would be on a newer version.

ivere27 replied:
Oh, bump the version(with changing the SIMPLE_HTTP_REQUEST_VERSION_... macros) as long as it reach 1.0.0 :)

ivere27 commented on CMakeLists.txt:
```diff
@@ -0,0 +1,49 @@
+# This file (c) 2017 AlertAvert.com.  All rights reserved.
```
AlertAvert.com ?

massenz replied:
Ha! (yes, my own domain :laughing: ) - removed.

ivere27 reviewed (commented):
could you check the comments?

massenz commented:
Thanks for super-quick reply!
Addressed comments and, in fact, commented out Conan in the cmake file; obviously, whether to remove or leave OpenSSL submodule is entirely up to you.

Different issues happen with `http` and `https` - I'll file a more detailed issue for that.
But using it against my own server worked great (I needed a client for my integration tests) and I'm super-happy with it, thanks!

Commit:
PR comments

ivere27 commented:
LGTM
